### PR TITLE
[Refactor] Refactor file writer for connector sink

### PR DESCRIPTION
**Pull request by letian-jiang**
## Why I'm doing:

Refactor file writer for connector sink. Part of https://github.com/StarRocks/starrocks/pull/40542

## What I'm doing:

- Define common interface `FileWriter` and `FileWriterFactory`
- Implement subclasses for Parquet and ORC

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/formats/parquet/parquet_file_writer.h [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    std::optional<std::vector<formats::FileColumnId>> _field_ids;
+    std::map<std::string, std::string> _options;
+    std::shared_ptr<ParquetWriterOptions> _parsed_options;
+
+    std::vector<std::string> _column_names;
+    std::vector<std::unique_ptr<ColumnEvaluator>> _column_evaluators;
+    PriorityThreadPool* _executors;
+};
+
+} // namespace starrocks::formats
```
[the remark this answers is not in the document]
Comment (letian-jiang) [reply be/src/formats/parquet/parquet_file_writer.h:130]: unique_ptr is used

File: be/src/formats/file_writer.h [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        std::string format;
+        FileMetrics file_metrics;
+        std::string location;
+        std::function<void()> rollback_action;
+    };
+
+    virtual ~FileWriter() = default;
+    virtual Status init() = 0;
+    virtual int64_t get_written_bytes() = 0;
+    virtual std::future<Status> write(ChunkPtr chunk) = 0;
```
Comment (letian-jiang) [inline be/src/formats/file_writer.h:56]: return `std::future` for async execution

File: be/src/util/blocking_priority_queue.hpp [hunk 3]
```diff
@@ -39,6 +39,7 @@
 #include <mutex>
 #include <vector>
 
+#include "common/compiler_util.h"
```
Comment (letian-jiang) [inline be/src/util/blocking_priority_queue.hpp:42]: define `ALWAYS_INLINE` in line 184

File: be/src/exec/orc_writer.h
Comment (letian-jiang) [inline be/src/exec/orc_writer.h:1]: the writer to put to `formats/orc/orc_file_writer.h`

File: be/src/bench/CMakeLists.txt [hunk 4]
```diff
@@ -19,10 +19,7 @@ endif()
 ADD_BE_BENCH(${SRC_DIR}/bench/chunks_sorter_bench)
 ADD_BE_BENCH(${SRC_DIR}/bench/runtime_filter_bench)
 ADD_BE_BENCH(${SRC_DIR}/bench/csv_reader_bench)
-ADD_BE_BENCH(${SRC_DIR}/bench/orc_writer_bench)
```
Comment (letian-jiang) [inline be/src/bench/CMakeLists.txt:22]: remove these benches for they are pointless now

**Comment by letian-jiang** [comment]
image

File: be/src/formats/file_writer.h [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+namespace starrocks::formats {
+
+struct FileWriterOptions {
+    virtual ~FileWriterOptions() = default;
+};
+
+class FileWriter {
+public:
+    struct FileMetrics {
```
[the remark this answers is not in the document]
Comment (letian-jiang) [reply be/src/formats/file_writer.h]: renamed

**Review by dirtysalt — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Pull request merged**